app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js

```diff
@@ -0,0 +1,100 @@
+import { useMemo, useCallback } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+
+import { dataUnionSecretSchema, dataUnionSecretsSchema } from '$shared/modules/entities/schema'
+
+import { setDataUnionSecrets, addDataUnionSecret, removeDataUnionSecret } from '../actions'
+import { selectDataUnionSecrets } from '../selectors'
+import { getSecrets, postSecret, putSecret, deleteSecret } from '../services'
+import useEntities from '$shared/hooks/useEntities'
+
+function useDataUnionSecrets() {
+    const dispatch = useDispatch()
+    const secrets = useSelector(selectDataUnionSecrets)
+    const { update } = useEntities()
+
+    const load = useCallback(async (dataUnionId) => {
+        try {
+            const response = await getSecrets({
+                dataUnionId,
```

Comment:
When you deploy a fresh DU, `dataUnionId` is `null`.

Reply:
Fixed in https://github.com/streamr-dev/streamr-platform/pull/1148/commits/ed2ffdf103711498773805fde647abbb31b3d61b

The effect reacted to the changing `beneficiaryAddress`, we should only load it when entering the page. Also added a check for join part stream creation fail.